utils/json-utils/src/test/scala/pl/touk/nussknacker/engine/json/swagger/extractor/JsonToNuStructTest.scala

```diff
@@ -76,13 +78,41 @@ class JsonToNuStructTest extends AnyFunSuite with Matchers {
   }
 
   test("should trim useless fields") {
-    val definition = SwaggerObject(elementType = Map("field3" -> SwaggerLong))
+    val definition = SwaggerObject(elementType = Map("field3" -> SwaggerLong), AdditionalPropertiesDisabled)
     extractor.JsonToNuStruct(json, definition) shouldBe TypedMap(Map.empty)
   }
 
+  test("should not trim useless fields when additionalPropertiesOn") {
+    val json = Json.obj("field1" -> fromString("value"), "field2" -> Json.fromInt(1))
+    val definition = SwaggerObject(elementType = Map("field3" -> SwaggerLong), AdditionalPropertiesWithoutType)
+    extractor.JsonToNuStruct(json, definition) shouldBe TypedMap(Map(
+      "field1" -> "value",
+      "field2"-> 1
+    ))
+
+    val jsonIntegers = Json.obj("field1" -> fromInt(2), "field2" -> fromInt(1))
+    val definition2 = SwaggerObject(elementType = Map("field3" -> SwaggerLong), AdditionalPropertiesSwaggerTyped(SwaggerLong))
+    extractor.JsonToNuStruct(jsonIntegers, definition2) shouldBe TypedMap(Map(
+      "field1" -> 2L,
+      "field2"-> 1L
+    ))
+  }
+
+  test("should throw exception on trying convert string to integer") {
+    val json = Json.obj("field1" -> fromString("value"), "field2" -> Json.fromInt(1))
+    val definition = SwaggerObject(elementType = Map("field3" -> SwaggerLong), AdditionalPropertiesSwaggerTyped(SwaggerLong))
+
+    intercept[JsonToObjectError] {
+      extractor.JsonToNuStruct(json, definition) shouldBe TypedMap(Map(
+        "field1" -> "value",
+        "field2"-> 1
+      ))
+    }
+  }
+
   test("should parse union") {
     val definition =
-      SwaggerObject(elementType = Map("field2" -> SwaggerUnion(List(SwaggerString, SwaggerLong))))
+      SwaggerObject(elementType = Map("field2" -> SwaggerUnion(List(SwaggerString, SwaggerLong))), AdditionalPropertiesWithoutType)
```

Comment:
maybe let's have default value `additionalProperties = AdditionalPropertiesBoolean(true)` as it is in JsonSchema specification?